diff --git a/projects/pluto/system_project.tcl b/projects/pluto/system_project.tcl
index 38531b11..25be482e 100644
--- a/projects/pluto/system_project.tcl
+++ b/projects/pluto/system_project.tcl
@@ -1,4 +1,3 @@
-
 source ../scripts/adi_env.tcl
 source $ad_hdl_dir/projects/scripts/adi_project.tcl
 source $ad_hdl_dir/projects/scripts/adi_board.tcl
@@ -11,7 +10,8 @@ adi_project_files pluto [list \
   "system_constr.xdc" \
   "$ad_hdl_dir/library/xilinx/common/ad_iobuf.v"]
 
+set_property include_dirs "$ad_hdl_dir/library/axi_dmac/" [current_fileset]
+
 set_property is_enabled false [get_files  *system_sys_ps7_0.xdc]
 adi_project_run pluto
 source $ad_hdl_dir/library/axi_ad9361/axi_ad9361_delay.tcl
-
