diff --git a/usrp3/tools/scripts/viv_generate_ip.tcl b/usrp3/tools/scripts/viv_generate_ip.tcl
index 8fe76933..30768ea1 100644
--- a/usrp3/tools/scripts/viv_generate_ip.tcl
+++ b/usrp3/tools/scripts/viv_generate_ip.tcl
@@ -20,6 +20,8 @@ file delete -force "$xci_file.out"
 create_project -part $part_name -in_memory -ip
 set_property target_simulator XSim [current_project]
 add_files -norecurse -force $xci_file
+report_ip_status
+upgrade_ip [get_ips *]
 reset_target all [get_files $xci_file]
 puts "BUILDER: Generating IP Target..."
 generate_target all [get_files $xci_file]
 
