diff --git a/Makefile b/Makefile
index 14beb67..9cb8000 100644
--- a/Makefile
+++ b/Makefile
@@ -21,7 +21,7 @@ SHELL = /bin/bash
 CC = $(CROSS_COMPILE)gcc
 
 # Standard gcc flags for compilation
-GLOBAL_CFLAGS = -Wall -Wextra -Werror -std=gnu99 -g -O0
+GLOBAL_CFLAGS = -Wall -Wextra -std=gnu99 -g -O0
 
 # The location where the compiled executables and driver will be stored
 OUTPUT_DIR ?= outputs
diff --git a/driver/Kbuild b/driver/Kbuild
index 0f89744..0aca1cb 100644
--- a/driver/Kbuild
+++ b/driver/Kbuild
@@ -20,4 +20,4 @@ obj-m += $(DRIVER_MODULE_NAME).o
 
 # Set the CFLAGS for compiling the module, adding the include flags and changing the default warning behavior.
 INC_FLAGS = $(addprefix -I ,$(DRIVER_INC_DIRS))
-ccflags-y = $(INC_FLAGS) -Werror -ggdb -std=gnu11
+ccflags-y = $(INC_FLAGS) -ggdb -std=gnu11
