diff --git a/driver/axidma_dma.c b/driver/axidma_dma.c
index 4b54de0..a041d38 100644
--- a/driver/axidma_dma.c
+++ b/driver/axidma_dma.c
@@ -143,7 +143,7 @@ static struct axidma_chan *axidma_get_chan(struct axidma_device *dev,
 static void axidma_dma_callback(void *data)
 {
     struct axidma_cb_data *cb_data;
-    struct siginfo sig_info;
+    struct kernel_siginfo sig_info;
 
     /* For synchronous transfers, notify the kernel thread waiting. For
      * asynchronous transfers, send a signal to userspace if requested. */
