diff --git a/linux/linux.mk b/linux/linux.mk
index 4b60f33ff3..9508e0aba1 100644
--- a/linux/linux.mk
+++ b/linux/linux.mk
@@ -381,6 +381,9 @@ ifeq ($(BR2_LINUX_KERNEL_DTS_SUPPORT),y)
 # shipped with the kernel sources, so we need flex and bison. For
 # reproducibility, we use our owns rather than the host ones.
 LINUX_DEPENDENCIES += host-bison host-flex
+ifeq ($(BR2_PACKAGE_AD9361_DTS),y)
+LINUX_DEPENDENCIES += ad9361-dts
+endif
 
 ifeq ($(BR2_LINUX_KERNEL_DTB_IS_SELF_BUILT),)
 define LINUX_BUILD_DTB
